### PR TITLE
Enhancement on CFGArg

### PR DESCRIPTION
**Pull request by chungshien**
> ### Motivate of the pull request
> - [x] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
Minor change in FOEDAG/src/Configuration/CFGCommon which greatly reduce dependency of CFGArg of ConfigurationRS from Configuration - In the future if we need to modify or add new CFGArg for ConfigurationRS, we do not need to touch Configuration at all (make the pull request much more easy)

> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [x Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: src/Configuration/CFGCommon/CFGCommon.cpp [hunk 1]
```diff
@@ -110,6 +115,14 @@ void CFG_post_msg(const std::string& message) {
   }
 }
 
+void CFG_post_warning(const std::string& message) {
+  if (m_msg_function != nullptr) {
+    m_msg_function(message);
+  } else {
+    printf("Warn : %s\n", message.c_str());
```
Comment (ravic-rs) [inline src/Configuration/CFGCommon/CFGCommon.cpp:122]: Why not use the python logging module here ?

Comment (chungshien) [reply src/Configuration/CFGCommon/CFGCommon.cpp:122]: Hi Ravi, can you elaborate more on python logging module? Maybe point to an example. Is it similar to FOEDAG::Compiler::Message/ErrorMessage()? Or there is other printing msg API?

Comment (chungshien) [reply src/Configuration/CFGCommon/CFGCommon.cpp:122]: Hi Ravi, can you elaborate more on python logging module? Maybe point to an example, thanks.
Is it similar to FOEDAG::Compiler::Message/ErrorMessage()? Or there is other printing msg API? 

BTW, this is c++ code, I am not sure how it is related to Python:
a. maybe we really support Python module in c++ which I am not aware of
b. maybe you mistakenly think this is .py file

Comment (ravic-rs) [reply src/Configuration/CFGCommon/CFGCommon.cpp:122]: Had a convo, we don't need to use compiler logging module here or in python code.

**Pull request merged**